pkg/jobstore/boltdb/context.go

```diff
@@ -0,0 +1,66 @@
+package boltjobstore
+
+import (
+	"context"
+
+	"github.com/bacalhau-project/bacalhau/pkg/jobstore"
+	bolt "go.etcd.io/bbolt"
+)
+
+// contextKey is a custom type to avoid key collisions in context values.
+type contextKey int
+
+// txContextKey is the key used to store the transaction context in the context.
+const txContextKey contextKey = 0
```

Comment:
```suggestion
// contextKey is a custom type to avoid key collisions in context values.
type contextKey struct{}

// txContextKey is the key used to store the transaction context in the context.
var txContextKey = contextKey{}
```
> WithValue returns a copy of parent in which the value associated with key is val.
> Use context Values only for request-scoped data that transits processes and APIs, **not for passing optional parameters to functions**.
> The provided key must be comparable and should not be of type string or any other built-in type to avoid collisions between packages using context. Users of WithValue should define their own types for keys. To avoid allocating when assigning to an interface{}, context keys often have concrete type struct{}. Alternatively, exported context key variables' static type should be a pointer or interface. 

https://pkg.go.dev/context#WithValue

Reply:
Yeah I am aware of the recommendation to not use built-in types to avoid collisions, and thats why I introduced a custom `contextKey` type. It doesn't have to be a struct
https://go.dev/blog/context#TOC_3.2.:~:text=Print(err)%0A%20%20%20%20%20%20%20%20return%0A%20%20%20%20%7D-,Package%20userip,-The%20userip%20package

Playground verification
https://go.dev/play/p/py6iZ6uL1BK

Reply:
fair enough. It may be worth maintaining consistency with other usages of contextKeys ([1](https://github.com/bacalhau-project/bacalhau/blob/fb9a5461b016e6016e4c31069bf08631b398b59f/cmd/util/cleanup.go#L9), [2](https://github.com/bacalhau-project/bacalhau/blob/fb9a5461b016e6016e4c31069bf08631b398b59f/cmd/cli/root.go#L192)) in the codebase, but will leave it to you.